### PR TITLE
Upgrade self to sbt 1.2.1

### PR DESCRIPTION
**Pull request by jeffmay**
@rallyhealth/engineers @htmldoug @rcmurphy @arkban @usufruct99 

- Fixes a regression when introducing a new artifact under the cover of `semVerEnforceAfterVersion`
- Fixes a minor regression in 1.1.0 where `semVerCheck` fails on projects that do not have any files in `src/main/scala`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ There are two sbt plugins in this one plugin library:
 
 # Compatibility
 
-Tested and supported for sbt version 0.13.x and above.
+Tested and supported for sbt versions: 0.13.17, 1.1.6, and 1.2.1
```
Comment (jeffmay) [inline README.md]: Technically this is what we test with. I'm assuming these tests reflect the other version in each minor version.

File: src/main/scala/com/rallyhealth/sbt/util/NullSbtLogger.scala [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
-  override def control(event: ControlEvent.Value, message: ⇒ String): Unit = ()
-
-  override def log(level: Level.Value, message: ⇒ String): Unit = ()
-
-  override def logAll(events: Seq[LogEvent]): Unit = ()
-
-  override def success(message: ⇒ String): Unit = ()
-
-  override def trace(t: ⇒ Throwable): Unit = ()
-}
```
Comment (jeffmay) [inline src/main/scala/com/rallyhealth/sbt/util/NullSbtLogger.scala]: Duplicate of test code version.

File: project/GitVersioningScriptedPlugin.scala [hunk 3]
```diff
@@ -1,337 +0,0 @@
...
-//        Try(sideChannelFile.deleteOnExit())
-//
-//        val sideChannelFileWriter = new PrintWriter(new FileWriter(sideChannelFile, false))
-//        sideChannelFileWriter.println(bufferedFile.getAbsolutePath)
-//        sideChannelFileWriter.close()
-//
-//        sideChannelFile
-//      }
-//  }
-//}
```
Comment (jeffmay) [inline project/GitVersioningScriptedPlugin.scala]: This was a hangover from long ago.

File: .travis.yml [hunk 4]
```diff
@@ -5,4 +5,4 @@ scala:
 jdk:
   - openjdk8
 script:
-  - sbt +test +scripted
+  - sbt ^test ^scripted
```
Comment (arkban) [inline .travis.yml]: WAT?

**Pull request merged**